packages/grafana-data/src/utils/url.ts

```diff
@@ -236,3 +237,21 @@ export const toURLRange = (range: RawTimeRange): URLRange => {
     to,
   };
 };
+
+/**
+ * Converts an object into a URL-encoded query string.
+ *
+ * @param params data to serialize
+ * @returns A URL-encoded string representing the provided data.
+ */
+function serializeParams(params: Record<string, string | number | boolean | Array<string | number | boolean>>): string {
```

Comment:
isn't this already covered by the function `toUrlParams` above?

Reply:
The `toUrlParams` does a lot of extra things and it also encodes values using `encodeURIComponentAsAngularJS` which I am not sure why is needed as we don't use angular anymore, but I am not sure as I don't have context if that is needed/redundant.

Reply:
yeah, I mean it does extra things but it seems it covers this use case. It would be nice to have some tests of what you expect with this function and checks if `toUrlParams` also passes them. If not it would be difficult for people to know which one to use.

Reply:
If `encodeURIComponentAsAngularJS` is the problem, you can always extend `toUrlParams` to optionally avoid that.

Reply:
I could do something like `function toUrlParams(a: any, useEncodeURIComponentAsAngularJS = true)`, but I think we still need to `useEncodeURIComponentAsAngularJS = true` for backward compatibility. Or do you think we can default to `false`? Or just remove `encodeURIComponentAsAngularJS`. 

@torkelo you worked on this years ago and you probably have the most context on if `encodeURIComponentAsAngularJS` is still needed. And if so, should we default `useEncodeURIComponentAsAngularJS = true` or `false`. Or if it should be dependant on `angular_support_enabled`?

Reply:
yes, I'd do it without the breaking change. You can also expose the functionality under a different name (e.g. `serializeParams`) but executing the same code and deprecate `toUrlParams`.

Or if we just want people to use this new function, simply deprecate the other one. At least the decission is clear about which one to use for new code.

Reply:
Hey @andresmgot, I've implemented a logic update. Now, we have a new function called `serializeParams` which utilizes `toUrlParams` but without AngularJS encoding. I've included comments to clarify the functionality of these functions. Regarding the `toUrlParams` function, I refrained from deprecating it since I lack information about its purpose, specifically why AngularJS encoding was necessary and whether it's still required. I'll leave the decision on deprecation to the team responsible for maintaining the code.